toolchain/sem_ir/inst_namer.cpp

```diff
@@ -441,6 +441,11 @@ auto InstNamer::CollectNamesInBlock(ScopeId scope_id,
         add_inst_name_id(sem_ir_.classes().Get(inst.class_id).name_id, ".type");
         continue;
       }
+      case CARBON_KIND(GenericInterfaceType inst): {
+        add_inst_name_id(sem_ir_.interfaces().Get(inst.interface_id).name_id,
```

Comment:
FWIW, it seems like both this and the formatter could benefit from a generic GetNameId(inst) kind of function. There's already a little of that in import.cpp, which also includes the parent scope, maybe we should be thinking about moving that to be a helper in sem_ir?

I think this just adds to the discussion we've had about wondering if there should be better helpers for common features of class/function/interface, though.

(just commenting, out of scope for this change)

Reply:
Yeah, more work to factor out the common parts would be great.